### PR TITLE
[eslint config] Prohibit global isNaN, isFinite

### PR DESCRIPTION
**Pull request by dlwalsh**
Fixes https://github.com/airbnb/javascript/issues/1347.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3107,16 +3137,16 @@ Other Style Guides
...
-  - [27.1](#es5-compat--kangax) Refer to [Kangax](https://twitter.com/kangax/)’s ES5 [compatibility table](https://kangax.github.io/es5-compat-table/).
+  <a name="es5-compat--kangax"></a><a name="28.1"></a>
+  - [28.1](#es5-compat--kangax) Refer to [Kangax](https://twitter.com/kangax/)’s ES5 [compatibility table](https://kangax.github.io/es5-compat-table/).
 
 **[⬆ back to top](#table-of-contents)**
 
 <a name="ecmascript-6-styles"></a>
 ## ECMAScript 6+ (ES 2015+) Styles
 
-  <a name="es6-styles"></a><a name="27.1"></a>
```
Comment (ljharb) [inline README.md]: all pre-existing numbered links must never be changed, so as to preserve old links.

File: README.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ Other Style Guides
   1. [Naming Conventions](#naming-conventions)
   1. [Accessors](#accessors)
   1. [Events](#events)
+  1. [Standard Library](#standard-library)
```
Comment (ljharb) [inline README.md]: to avoid renumbering all the rest of the sections, let's just add this at the end.

File: README.md [hunk 3]
```diff
@@ -3036,10 +3037,39 @@ Other Style Guides
 
   **[⬆ back to top](#table-of-contents)**
 
+## Standard library
+
+  The [Standard Library](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects)
+  contains utilities that are functionally broken but remain for legacy reasons.
+
+  <a name="standard-library--isnan"></a><a name="26.1"></a>
```
Comment (ljharb) [inline README.md]: numbered links only live on legacy items; new items should not have them.

File: packages/eslint-config-airbnb-base/rules/best-practices.js [hunk 4]
```diff
@@ -186,13 +186,24 @@ module.exports = {
...
+    'no-restricted-globals': ['error', 'isFinite', 'isNaN'],
+
     // disallow certain object properties
     // http://eslint.org/docs/rules/no-restricted-properties
     'no-restricted-properties': ['error', {
       object: 'arguments',
       property: 'callee',
       message: 'arguments.callee is deprecated',
     }, {
+      object: 'window',
```
Comment (ljharb) [inline packages/eslint-config-airbnb-base/rules/best-practices.js]: we should do the same for `global.isFinite`, and `self.isFinite`.

**Comment by dlwalsh** [comment]
Requested changes done.

1. Four extra entries for `global.isFinite`, `self.isFinite`, `global.isNaN`, `self.isNaN`.
2. README: Standard Library now section 29. Testing bumped to 30. All links restored to master versions.

**Comment by icywolfy** [comment]
Just want to mention here for anyone who reads the Changelog:
If you target IE:  Neither IE nor IE/Mobile support Number.isFinite, or Number.isNaN

**Comment by ljharb** [comment]
@icywolfy See the note at the top of the guide: even IE 6 supports these things if you're using airbnb-browser-shims.

**Pull request merged**